### PR TITLE
[LC-624] Remove `REDIRECT_PROTOCOL` and its followings

### PR DESCRIPTION
**Pull request by metalg0su**
# Goal
- Remove unused Option `REDIRECT_PROTOCOL`
> Now relay target is supplied from loopchain engine. So remove it to avoid possible confusions.

### CODE REVIEW

File: tests/dispatcher/test_dispatcher.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+from iconrpcserver.server.rest_property import RestProperty
+from iconrpcserver.utils import json_rpc
+
+from iconrpcserver.utils.message_queue.stub_collection import StubCollection, IconScoreInnerStub
+
+CHANNEL_NAME = "icon_dex"
+
+
+@pytest.fixture(autouse=True)
+def mocking_rest_property():
```
Comment (laurenceyoon) [inline tests/dispatcher/test_dispatcher.py]: Why `mocking_rest_property` function is defined twice?

File: tests/dispatcher/test_dispatcher.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    rest_prop.relay_target = {CHANNEL_NAME: "relay_target"}
+
+    yield rest_prop
+
+    RestProperty().rs_target = None
+    RestProperty().relay_target = None
+
+
+@pytest.fixture(autouse=True)
+def mocking_rest_property():
```
Comment (metalg0su) [inline tests/dispatcher/test_dispatcher.py]: change method name!

**Comment by metalg0su** [comment]
redirectProtocol option should be removed if LC-513 merged. (Add snapcraft)
> https://github.com/icon-project/loopchain/pull/448#discussion_r362685034
```
- citizen/conf/iconrpcserver_mainnet.json, 
- citizen/conf/iconrpcserver_testnet.json
```

File: iconrpcserver/utils/json_rpc.py [hunk 3]
```diff
@@ -93,8 +90,12 @@ def process_response(self, response, log_extra=None, log_format=None):
                     f"relay_target[{relay_target}], "
                     f"version[{version}], "
                     f"method[{method_name}]")
-        result = await CustomAiohttpClient(session, relay_uri).request(method_name, message)
+        print("type: ", CustomAiohttpClient.request)
+        client = CustomAiohttpClient(session, relay_uri)
+        print("cli: ", client)
+        result = await client.request(method_name, message)
         Logger.debug(f"relay_tx_request result[{result}]")
+        print(f"relay_tx_request result[{result}]")
```
Comment (yakkle) [inline iconrpcserver/utils/json_rpc.py]: Is this draft?
I think this lines need to update.

Comment (metalg0su) [reply iconrpcserver/utils/json_rpc.py]: I made a mistake.
I will remove debug lines.

Comment (metalg0su) [reply iconrpcserver/utils/json_rpc.py]: Sorry to be late!

**Review by windies21 — APPROVED** [review]

File: tests/dispatcher/test_dispatcher.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        async def validate_transaction(self, request):
+            return "response"
+
+    StubCollection().conf = {
+        "channel": CHANNEL_NAME
+    }
+    StubCollection().icon_score_stubs[CHANNEL_NAME] = MockIconScoreInnerStub(MockIconScoreInner)
+    StubCollection().channel_tx_creator_stubs[CHANNEL_NAME] = MockChannelTxCreatorStub(MockChannelTxCreator)
+
+    yield
```
Comment (yakkle) [inline tests/dispatcher/test_dispatcher.py:58]: What purpose of this `yield`?

File: tests/dispatcher/test_dispatcher.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+class MockChannelTxCreatorStub:
+    def __init__(self, mock_cls):
+        self.mock_class = mock_cls
+
+    def async_task(self):
+        return self.mock_class()
+
+
+@pytest.mark.asyncio
+class TestDispatcher_v2:
```
Comment (yakkle) [inline tests/dispatcher/test_dispatcher.py]: Is this classname follow convention rule?

File: tests/dispatcher/test_dispatcher.py (hunk 4 above)
Comment (metalg0su) [reply tests/dispatcher/test_dispatcher.py:58]: Tests with fixtures run at following order:
- call fixture
- run test

but if `yield` exists in fixture:
- call fixture 
- stop at `yield`
- run test
- after test, run fixture after `yield` line.

It works like contextmanager.
I used it to get the same effects of `setup` and `teardown` in `unittest`.

File: tests/dispatcher/test_dispatcher.py (hunk 5 above)
Comment (metalg0su) [reply tests/dispatcher/test_dispatcher.py]: Do I use `TestDispatcherV2` or `TestDispatcherv2`?

Hmm.. I think former is better.
I will fix the name.

**Review by yakkle — APPROVED** [review]

**Review by zsaladin — APPROVED** [review]

**Pull request merged**